File: televisionbingecalculator/src/main/java/com/vanillax/televisionbingecalculator/app/TBC/adapters/StreamingSourceRecyclerAdapter.java
package com.vanillax.televisionbingecalculator.app.TBC.adapters;

import com.vanillax.televisionbingecalculator.app.R;
import com.vanillax.televisionbingecalculator.app.ServerAPI.movie.Offer;
import com.vanillax.televisionbingecalculator.app.Util.BindingAdapter.BaseDataBindingAdapter;
import com.vanillax.televisionbingecalculator.app.Util.BindingAdapter.DataBoundViewHolder;
import com.vanillax.televisionbingecalculator.app.databinding.StreamingSourceBinding;
import com.vanillax.televisionbingecalculator.app.viewmodel.StreamingSourceViewModelItem;

import java.util.ArrayList;
import java.util.List;

import static com.vanillax.televisionbingecalculator.app.ServerAPI.movie.Offer.AMAZON;
import static com.vanillax.televisionbingecalculator.app.ServerAPI.movie.Offer.HBO;
import static com.vanillax.televisionbingecalculator.app.ServerAPI.movie.Offer.HULU;
import static com.vanillax.televisionbingecalculator.app.ServerAPI.movie.Offer.NETFLIX;
import static com.vanillax.televisionbingecalculator.app.ServerAPI.movie.Offer.VUDU;

/**
 * Created by mitchross on 12/9/14.
 */
public class StreamingSourceRecyclerAdapter extends BaseDataBindingAdapter<StreamingSourceBinding>
{

   private List<StreamingSourceViewModelItem> streamingSourceViewModelItems;


    public StreamingSourceRecyclerAdapter()
    {
        streamingSourceViewModelItems = new ArrayList<>(  );
    }

    public void setStreamingSourceViewModelItems( List<Offer> movieOffers )
    {
        boolean netflix = false;
        boolean hulu = false;
        boolean vudu = false;
        boolean hbo = false;
        boolean amazon = false;


        streamingSourceViewModelItems.clear();
        for ( Offer offer : movieOffers)
        {

            if( offer.providerId != 0 )
            {

                switch ( offer.providerId )
                {
                    case NETFLIX:
                        if ( !netflix )
                        {
                            this.streamingSourceViewModelItems.add( new StreamingSourceViewModelItem( StreamingSourceViewModelItem.NETFLIX ) );
                            netflix = true;
                        }
                        break;
                    case HULU:
                        if (!hulu)
                        {
                            this.streamingSourceViewModelItems.add( new StreamingSourceViewModelItem( StreamingSourceViewModelItem.HULU ) );
                            hulu = true;
                        }
                        break;
                    case VUDU:
                        if(!vudu)
                        {
                            this.streamingSourceViewModelItems.add( new StreamingSourceViewModelItem( StreamingSourceViewModelItem.VUDU ) );
                            vudu = true;
                        }
                        break;
                    case HBO:
                        if (!hbo)
                        {
                            this.streamingSourceViewModelItems.add( new StreamingSourceViewModelItem( StreamingSourceViewModelItem.HBO ) );
                            hbo = true;
                        }
                        break;
                    case AMAZON:
                        if(!amazon)
                        {
                            this.streamingSourceViewModelItems.add( new StreamingSourceViewModelItem( StreamingSourceViewModelItem.AMAZON ) );
                            amazon = true;
                        }
                        break;


                }
            }
        }
        notifyDataSetChanged();

    }

    @Override
    protected void bindItem( DataBoundViewHolder<StreamingSourceBinding> holder, int position, List<Object> payloads )
    {
        holder.binding.setViewModel( streamingSourceViewModelItems.get( position ) );
    }

    @Override
    public int getItemCount()
    {
        return streamingSourceViewModelItems.size();
    }

    @Override
    public int getItemViewType( int position )
    {
        return R.layout.streaming_source;
    }




}

File: televisionbingecalculator/src/main/java/com/vanillax/televisionbingecalculator/app/tbc/adapters/StreamingSourceRecyclerAdapter.kt
package com.vanillax.televisionbingecalculator.app.tbc.adapters

import androidx.recyclerview.widget.DiffUtil
import com.vanillax.televisionbingecalculator.app.R
import com.vanillax.televisionbingecalculator.app.kotlin.network.response.Offer
import com.vanillax.televisionbingecalculator.app.util.bindingadapter.BaseDataBindingAdapter
import com.vanillax.televisionbingecalculator.app.util.bindingadapter.BaseDataBindingListAdapter
import com.vanillax.televisionbingecalculator.app.util.bindingadapter.DataBoundViewHolder
import com.vanillax.televisionbingecalculator.app.viewmodel.SeasonNumberViewModelItem
import com.vanillax.televisionbingecalculator.app.viewmodel.StreamingSourceViewModelItem
import java.util.*

/**
 * Created by mitchross on 12/9/14.
 */
class StreamingSourceRecyclerAdapter : BaseDataBindingAdapter<com.vanillax.televisionbingecalculator.app.databinding.StreamingSourceBinding>() {

    private val streamingSourceViewModelItems: MutableList<StreamingSourceViewModelItem>

    init {
        streamingSourceViewModelItems = ArrayList()
    }

    fun setStreamingSourceViewModelItems(movieOffers: List<Offer>) {
        var netflix = false
        var hulu = false
        var vudu = false
        var hbo = false
        var amazon = false


        streamingSourceViewModelItems.clear()
//        for (offer in movieOffers) {
//
//            if (offer.providerId != 0) {
//
//                when (offer.providerId) {
//                    Offer.Streamtype.NETFLIX.value -> if (!netflix) {
//                        this.streamingSourceViewModelItems.add(StreamingSourceViewModelItem(StreamingSourceViewModelItem.NETFLIX))
//                        netflix = true
//                    }
//                    Offer.Streamtype.HULU.value -> if (!hulu) {
//                        this.streamingSourceViewModelItems.add(StreamingSourceViewModelItem(StreamingSourceViewModelItem.HULU))
//                        hulu = true
//                    }
//                    Offer.Streamtype.VUDU.value -> if (!vudu) {
//                        this.streamingSourceViewModelItems.add(StreamingSourceViewModelItem(StreamingSourceViewModelItem.VUDU))
//                        vudu = true
//                    }
//                    Offer.Streamtype.HBO.value -> if (!hbo) {
//                        this.streamingSourceViewModelItems.add(StreamingSourceViewModelItem(StreamingSourceViewModelItem.HBO))
//                        hbo = true
//                    }
//                    Offer.Streamtype.AMAZON.value -> if (!amazon) {
//                        this.streamingSourceViewModelItems.add(StreamingSourceViewModelItem(StreamingSourceViewModelItem.AMAZON))
//                        amazon = true
//                    }
//                }
//            }
//        }
        notifyDataSetChanged()

    }



    override fun bindItem(holder: DataBoundViewHolder<com.vanillax.televisionbingecalculator.app.databinding.StreamingSourceBinding>, position: Int, payloads: List<Any>) {
        holder.binding.viewModel = streamingSourceViewModelItems[position]
    }

    override fun getItemCount(): Int {
        return streamingSourceViewModelItems.size
    }

    override fun getItemViewType(position: Int): Int {
        return R.layout.streaming_source

    }
}

class StreamingSourceRecyclerAdapter2 : BaseDataBindingListAdapter<StreamingSourceViewModelItem>(DiffUtilItemCallback()) {

    override fun getItemViewType(position: Int): Int {
        return getItem(position).viewType
    }

    class DiffUtilItemCallback: DiffUtil.ItemCallback<StreamingSourceViewModelItem>() {

        override fun areItemsTheSame(oldItem: StreamingSourceViewModelItem,
                                     newItem: StreamingSourceViewModelItem): Boolean {
            return oldItem.streamingIcon == newItem.streamingIcon
        }

        override fun areContentsTheSame(oldItem: StreamingSourceViewModelItem,
                                        newItem: StreamingSourceViewModelItem): Boolean {
            return true
        }
    }
}
